.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,266 @@
+        run: |
+          echo "Linting composer.json"
+          jq empty composer.json
+          for JSON in .github/workflows/*.json;do
+            echo "Linting $JSON"
+            jq empty $JSON
+          done
+
+  dependency-versions-matrix:
+    name: Dependencies to Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.dependency-versions-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: dependency-versions-matrix
+        run: |
+          DEPS=$(if [ -f composer.lock ]; then echo -n '["lowest", "locked", "latest"]'; else echo -n '["lowest", "latest"]'; fi)
+          echo "Testing against dependencies: ${DEPS}"
+          echo "::set-output name=check::${DEPS}"
+
+  config:
+    name: Retrieve configuration
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      json: ${{ steps.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: config
+        run: |
+          CONFIG=$(if [ -f ${CONFIG_FILE} ]; then jq -c '.' < ${CONFIG_FILE} ; else echo -n "{}" ; fi)
+          echo "Using config: $CONFIG"
+          echo "::set-output name=json::${CONFIG}"
+
+  supported-versions-matrix:
+    name: Supported Versions Matrix
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      version: ${{ steps.supported-versions-matrix.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-versions-matrix
+        uses: WyriHaximus/github-action-composer-php-versions-in-range@v1
+
+  supported-checks-matrix:
+    name: Determine which checks to run
+    runs-on: ubuntu-latest
+    needs: [config]
+    outputs:
+      check: ${{ steps.supported-checks-matrix.outputs.check }}
+    env:
+      CONFIG_JSON: ${{ needs.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-checks-matrix
+        run: |
+          # This grabs the default known checks from the laminas.dev API.
+          #
+          # If the .github/workflows/continuous-integration-config.json file is
+          # present, and defines a "checks" array, that value will be
+          # used as the list of known checks.
+          #
+          # This list is then compared to the list of scripts in composer.json,
+          # and the intersection of them is provided as the output.
+          #
+          # If you want to use fewer or more checks than the defaults (e.g., to run
+          # benchmarks provided in the current repository, or NOT run static
+          # analysis, even if the scripts are present), create and/or update the
+          # continuous-integration-config.json file. The value should include all
+          # checks which should be run.
+          JQ_SCRIPT_TEMPLATE='def tools:TOOLS ; def intersect(x;y): (x + y | sort) as $sorted | reduce range(1; $sorted|length) as $i ([]; if $sorted[$i] == $sorted[$i-1] then . + [$sorted[$i]] else . end) ; .scripts | keys | intersect(.;tools)'
```

Comment:
I'm OK with hacking things into oneline for a single use-case, but this would be a serious maintenance problem if every repo had it :|

Reply:
We might be able to do multiple lines here; I was just unsure about the interaction of YAML and bash works exactly in GHA config. I can experiment with it some more... though after chatting with you on Slack, it may be irrelevant if we can move this into an action.